beacon-chain/sync/validate_beacon_blocks_test.go

```diff
@@ -1495,3 +1496,225 @@ func Test_validateDenebBeaconBlock(t *testing.T) {
 	require.NoError(t, err)
 	require.ErrorIs(t, validateDenebBeaconBlock(bdb.Block()), errRejectCommitmentLen)
 }
+
+func TestDetectAndBroadcastEquivocation_NoEquivocation(t *testing.T) {
```

Comment:
Can you merge all these tests into one, using `t.Run` for each test case? That way you can set up things like `p := p2ptest.NewTestP2P(t)` only once, at the start of the test, and reduce test times.

Pseudocode:
```
func TestDetectAndBroadcastEquivocation() {
    p := p2ptest.NewTestP2P(t)
    (...)
    t.Run("no equivocation", ...
    t.Run("equivocation detected", ...
}
```